core/policyToken.go

```diff
@@ -232,11 +265,35 @@ func vaultHealthCheck(config *tomlConfig, httpClient *http.Client) (sCode int, e
 	}
 	defer resp.Body.Close()
 
-	lc.Info(fmt.Sprintf("Vault Health Check HTTP Status: %s (StatusCode: %s)", resp.Status, strconv.Itoa(resp.StatusCode)))
+	lc.Info(fmt.Sprintf("Vault Policy Check HTTP Status: %s (StatusCode: %s)", resp.Status, strconv.Itoa(resp.StatusCode)))
 
 	return resp.StatusCode, nil
 }
 
+// ----------------------------------------------------------
+func hashFile(policyFilePtr *string, debug bool) (hashSum []byte, err error) {
+
+	inputFile, err := os.Open(*policyFilePtr)
+	if err != nil {
+		lc.Error(fmt.Sprintf("Fatal Error opening the Vault policy file: %s", err))
+		return nil, err
+	}
+	defer inputFile.Close()
+
+	hash256 := sha256.New()
+	if _, err := io.Copy(hash256, inputFile); err != nil {
+		lc.Error(fmt.Sprintf("Fatal Error calculating the SHA256 hash of the Vault policy file: %s", err))
+		return nil, err
+	}
+
+	hashSum = hash256.Sum(nil)
+	if debug {
+		lc.Info(fmt.Sprintf("Vault policy file checksum (SHA256): %x", hashSum))
```

Comment:
again, I think this should ideally be at debug level for the logs, but since it's behind a debug flag it's probably okay